src/charconv.c

```diff
@@ -29,7 +29,7 @@ static int init_conversion(int codepage)
 	initialized = 1;
 	if (codepage < 0)
 	    codepage = DEFAULT_DOS_CODEPAGE;
-	setlocale(LC_ALL, "");	/* initialize locale */
+	setlocale(LC_CTYPE, "");	/* initialize locale for CODESET */
```

Comment:
Is there a reason to change this? If it's just "it's not strictly needed to do `LC_ALL`" then I would prefer to leave it as it is.

Otherwise the changes look good.

Reply:
LC_ALL includes also LC_MESSAGES. And LC_MESSAGES is used by functions like strerror(). Therefore if you have system locale set to non-English then you will have partially translated messages, e.g.: https://github.com/dosfstools/dosfstools/blob/bfeedfce3a702a04eb8ad302d40aff623445e931/src/io.c#L149-L151

When you initialize just LC_CTYPE, then LC_MESSAGES stays in its default (C locale) and so error messages printed by strerror would be in C locale (in English) despite how is LC_ALL or LC_MESSAGES env variable set.